src/input_common/analog_from_button.cpp

```diff
@@ -11,31 +14,103 @@ class Analog final : public Input::AnalogDevice {
     using Button = std::unique_ptr<Input::ButtonDevice>;
 
     Analog(Button up_, Button down_, Button left_, Button right_, Button modifier_,
-           float modifier_scale_)
+           float modifier_scale_, float modifier_angle_)
         : up(std::move(up_)), down(std::move(down_)), left(std::move(left_)),
-          right(std::move(right_)), modifier(std::move(modifier_)),
-          modifier_scale(modifier_scale_) {}
-
-    std::tuple<float, float> GetStatus() const override {
-        constexpr float SQRT_HALF = 0.707106781f;
-        int x = 0, y = 0;
+          right(std::move(right_)), modifier(std::move(modifier_)), modifier_scale(modifier_scale_),
+          modifier_angle(modifier_angle_) {
+        update_thread = std::thread(&Analog::UpdateStatus, this);
+    }
 
-        if (right->GetStatus()) {
-            ++x;
+    ~Analog() override {
+        update_thread_running = false;
+        if (update_thread.joinable()) {
+            update_thread.join();
         }
-        if (left->GetStatus()) {
-            --x;
-        }
-        if (up->GetStatus()) {
-            ++y;
+    }
+
+    void MoveToDirection(bool enable, float to_angle) {
+        constexpr float TAU = 3.1415926f * 2.0f;
```

Comment:
make use of the Common::PI constant in `common/math_util.h`
```suggestion
        constexpr float TAU = Common::PI * 2.0f;
```